apps/oc_chef_wm/src/oc_chef_object_db.erl

```diff
@@ -117,6 +117,12 @@ maybe_delete_authz_id_or_error(1, Object, _RequestorId)  when is_record(Object,
                                                               is_record(Object, oc_chef_org_user_invite) ->
     %% These objects have no authz id
     ok;
+maybe_delete_authz_id_or_error({ok, 1}, #oc_chef_cookbook_artifact_version{authz_id = AuthzId}, RequestorId) ->
+    case oc_chef_cookbook_artifact:exists_by_authz_id(AuthzId) of
```

Comment:
In the case of cookbook_versions, I believe it happens because the cookbook authz  id is shared across versions and I think cleanup occurs when the last version is deleted - not directly by authzid, but when we delete the cookbook itself as a result of deleting the last version. 

I've been context-switching too much lately - is this similar,  where we could have multiple cb artifacts using a shared authz id?